extensions/src/main/java/org/pentaho/platform/plugin/action/ActionResourceUtil.java

```diff
@@ -0,0 +1,72 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation..  All rights reserved.
+ */
+package org.pentaho.platform.plugin.action;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.platform.api.action.ActionInvocationException;
+import org.pentaho.platform.api.action.IAction;
+import org.pentaho.platform.api.action.IActionInvokeStatus;
+import org.pentaho.platform.api.action.IActionInvoker;
+import org.pentaho.platform.plugin.action.messages.Messages;
+import org.pentaho.platform.util.ActionUtil;
+import org.pentaho.platform.util.StringUtil;
+import org.slf4j.MDC;
+
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Optional;
+
+public class ActionResourceUtil {
```

Comment:
ActionUtil is in core, whereas this is in extensions, I think it will be helpful as most of the code for WN is in extensions, this code does not need to be accessed by all of core, so I created a second file, but I am happy to move it if ActionUtil is the better place